frame/assets/src/types.rs

```diff
@@ -185,3 +187,39 @@ impl From<TransferFlags> for DebitFlags {
 		}
 	}
 }
+
+/// Possible errors when converting between external and asset balances.
+#[derive(Eq, PartialEq, Copy, Clone, RuntimeDebug, Encode, Decode)]
+pub enum ConversionError {
+	/// The asset is not present in storage.
+	AssetMissing,
+	/// The asset is not sufficient and thus does not have a reliable `min_balance` so it cannot be converted.
+	AssetNotSufficient,
+}
+
+/// Converts a balance value into an asset balance based on the ratio between the existential
+/// deposit and the minimum asset balance.
+pub struct BalanceToAssetBalance<T, Balance, ED>(PhantomData<(T, Balance, ED)>);
+impl<T: Config, Balance, ED> BalanceToAssetBalance<T, Balance, ED>
+where
+	Balance: Into<<T as Config>::Balance> + FixedPointOperand + Ord + One,
+	<T as Config>::Balance: FixedPointOperand,
+	ED: Get<Balance>,
+{
+	/// Convert the given balance value into an asset balance based on the ratio between the existential
+	/// deposit and the minimum asset balance.
+	///
+	/// Will return `Err` if the asset is not found or not sufficient.
+	pub fn to_asset_balance(balance: Balance, asset_id: <T as Config>::AssetId) -> Result<<T as Config>::Balance, ConversionError> {
```

Comment:
if there's no trait to be fulfilled, you might as well just expose a function rather than a struct.